File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/UIntUniform.java
package org.saar.lwjgl.opengl.shaders.uniforms;

import org.lwjgl.opengl.GL30;

public abstract class UIntUniform extends UniformBase implements Uniform {

    @Override
    public final void doLoad() {
        GL30.glUniform1ui(getLocation(), getUniformValue());
    }

    public abstract int getUniformValue();
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/UIntUniform.kt
package org.saar.lwjgl.opengl.shaders.uniforms

import org.lwjgl.opengl.GL30

abstract class UIntUniform : Uniform {

    final override fun load(location: Int) {
        val value: Int = this.value
        GL30.glUniform1ui(location, value)
    }

    abstract val value: Int
}